contracts/bridge/contracts/binance/BasicOutboundChannel.sol

```diff
@@ -2,23 +2,37 @@
 pragma solidity >=0.6.0 <0.7.0;
 pragma experimental ABIEncoderV2;
 
+import "@openzeppelin/contracts/access/AccessControl.sol";
 import "../interfaces/IOutboundChannel.sol";
 
 // BasicOutboundChannel is a basic channel that just sends messages with a nonce.
-contract BasicOutboundChannel is IOutboundChannel {
+contract BasicOutboundChannel is IOutboundChannel, AccessControl {
 
-    uint64 public nonce;
+    bytes32 public constant OUTBOUND_ROLE = keccak256("OUTBOUND_ROLE");
 
+    uint256 public nonce;
+
+    /**
+     * The Message is the structure of EthereRPC which should be delivery to Darwinia
+     * @param source The source sender address which send the message
+     * @param nonce The ID used to uniquely identify the message
+     * @param payload The calldata which encoded by ABI Encoding `abi.encodePacked(SELECTOR, PARAMS)`
+     */
     event Message(
         address source,
-        uint64 nonce,
+        uint256 nonce,
         bytes payload
     );
 
+    constructor() public {
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
     /**
      * @dev Sends a message across the channel
      */
     function submit(bytes calldata payload) external override {
+        require(hasRole(OUTBOUND_ROLE, msg.sender), "Channel: not-authorized");
```

Comment:
Missing docs, and 1. why it requires auth here? 2. how the fees are paid?

Reply:
Related to the task `outbound channel add whitlist`.
At the beginning of the bridge, only whitelisted apps may be required to send messages from Ethereum to Darwinia.